alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractConfigResourceActions.java

```diff
@@ -0,0 +1,182 @@
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.action.api;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.http.HttpStatus;
+
+import com.synopsys.integration.alert.common.action.ActionResponse;
+import com.synopsys.integration.alert.common.action.ValidationActionResponse;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.persistence.accessor.DescriptorAccessor;
+import com.synopsys.integration.alert.common.persistence.model.RegisteredDescriptorModel;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
+
+public abstract class AbstractConfigResourceActions implements ResourceActions<FieldModel>, TestAction<FieldModel>, ValidateAction<FieldModel> {
+    private AuthorizationManager authorizationManager;
+    private DescriptorAccessor descriptorAccessor;
+
+    public AbstractConfigResourceActions(AuthorizationManager authorizationManager, DescriptorAccessor descriptorAccessor) {
+        this.authorizationManager = authorizationManager;
+        this.descriptorAccessor = descriptorAccessor;
+    }
+
+    public AbstractConfigResourceActions(AuthorizationManager authorizationManager) {
+        this.authorizationManager = authorizationManager;
+    }
+
+    protected abstract ActionResponse<FieldModel> createResource(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> deleteResource(Long id);
+
+    protected abstract ActionResponse<List<FieldModel>> readAllResources();
+
+    protected abstract ActionResponse<List<FieldModel>> readAllByContextAndDescriptor(String context, String descriptorName);
+
+    protected abstract Optional<FieldModel> findFieldModel(Long id);
+
+    protected abstract ValidationActionResponse testResource(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> updateResource(Long id, FieldModel resource);
+
+    protected abstract ValidationActionResponse validateResource(FieldModel resource);
```

Comment:
Looking at these names and how they are used below, we might want to distinguish them a little more from each other. Maybe instead of the public `create` and the protected `createResource`, the public method could be called `createWithPermissionCheck` and the protected method could be called `createWithoutPermissionCheck` or something along those lines.

Reply:
I would prefer to change the names of the abstract methods because the public methods are displayed in the Swagger API.  To an end user they just want to create, read, update, delete, validate, or test the resources. Whether we perform permission checks or not should not be exposed to them.

Reply:
Aren't the actions invoked by the controllers? The name of the method in the controller should be the name that is displayed, no?

Reply:
You're right been looking at this code for too long.